internal/terminal/table.go

```diff
@@ -0,0 +1,151 @@
+// data is a map of a string to another string to print, where the key should be a value in headers
+//	if not, we will not print
+// maxHeaderWidth is to determine the column width based on the max length of the values and header
+type table struct {
+	headers        []string
+	data           []map[string]string
+	maxHeaderWidth map[string]int
+}
+
+// columnBreakWidth is to set a standard for the break between columns for the table
+const (
+	 columnBreakWidth = 2
+)
+
+func newTable(headers []string, values []map[string]interface{}) *table {
+	if headers == nil || len(headers) == 0 ||
+		values == nil || len(values) == 0 {
+		return nil
+	}
+
+	maxHeaderWidth := make(map[string]int)
+	for _, h := range headers {
+		temp := len(h)
+		if temp > maxHeaderWidth[h] {
+			maxHeaderWidth[h] = temp
+		}
+	}
+
+	var stringValues []map[string]string
+	for _, v := range values {
+		tempVal := make(map[string]string)
+		for k1, v1 := range v {
+			// convert the interface values into strings for printing
+			var val string
+			switch v := v1.(type) {
+			case string:
+				spaceRegex := regexp.MustCompile(`\s+`)
+				val = spaceRegex.ReplaceAllString(v, " ")
+			case float32, float64:
+				val = fmt.Sprintf("%g", v)
+			case fmt.Stringer:
+				val = v.String()
+			case nil:
+				val = ""
+			default:
+				val = fmt.Sprintf("%v", v)
+			}
+			valLen := len(val)
+			if valLen > maxHeaderWidth[k1] {
+				maxHeaderWidth[k1] = valLen
+			}
+			tempVal[k1] = val
+		}
+		stringValues = append(stringValues, tempVal)
+	}
+
+	resultTable := &table{
+		headers,
+		stringValues,
+		maxHeaderWidth,
+	}
+	return resultTable
+}
+
+func (t table) writeHeader() string {
+	var tempBuilder strings.Builder
+	var result strings.Builder
+	for i, h := range t.headers {
+		spaceCount := t.maxHeaderWidth[h] - len(h) + columnBreakWidth
+		// If it's not the last column
+		if i + 1 >= len(t.headers) {
+			spaceCount -= columnBreakWidth
+		}
+		spaces := strings.Repeat(" ", spaceCount)
+		tempBuilder.WriteString("%s")
+		tempBuilder.WriteString(spaces)
+
+		result.WriteString(fmt.Sprintf(tempBuilder.String(), color.New(color.Bold).SprintFunc()(h)))
+		tempBuilder.Reset()
+	}
+	return result.String()
+}
+
+func (t table) writeValues() string {
+	var result strings.Builder
+	var tempBuilder strings.Builder
+
+	for _, valueMap := range t.data {
+		if len(valueMap) > 0 {
+			for i, header := range t.headers {
+				spaceCount := t.maxHeaderWidth[header] - len(valueMap[header]) + columnBreakWidth
+				if i + 1 >= len(t.headers) {
+					spaceCount -= columnBreakWidth
+				}
+				spaces := strings.Repeat(" ", spaceCount)
+
+				tempBuilder.WriteString("%s")
+				tempBuilder.WriteString(spaces)
+
+				result.WriteString(fmt.Sprintf(tempBuilder.String(), valueMap[header]))
+				tempBuilder.Reset()
+			}
+			result.WriteString("\n")
+		}
+	}
+	return result.String()
+}
+
+func (t table) writeDividers() string {
+	var divider strings.Builder
+	var tempBuilder strings.Builder
```

Comment:
same thing here, i'll let you try to work through this implementation where we're only needing a single `var builder strings.Builder`